lib/core/topologies/replset_state.js

```diff
@@ -358,7 +359,8 @@ ReplSetState.prototype.update = function(server) {
   // Standalone server, destroy and return
   //
   if (ismaster && ismaster.ismaster && !ismaster.setName) {
-    this.topologyType = this.primary ? TopologyType.ReplicaSetWithPrimary : TopologyType.Unknown;
+    // We should not mark the topology as Unknown because of one standalone
```

Comment:
again just curious what this fixes

Reply:
It fixed an SDAM test where if a node that is within a replicaset declares itself as a standalone it should be removed from the topology but the topology should not become Unknown, simply remain ReplicaSetNoPrimary. This test: (test/spec/server-discovery-and-monitoring/rs/member_standalone.json)